ibis/common/typing.py

```diff
@@ -259,3 +260,21 @@ class Coercible(Abstract):
     @classmethod
     @abstractmethod
     def __coerce__(cls, value: Any, **kwargs: Any) -> Self: ...
+
+
+def get_defining_frame(obj):
```

Comment:
Perhaps not the typing module is the best place for this, but we don't have an inspect module here yet. We can reorganize later since these are designed for public use.